subscribe.go

```diff
@@ -85,3 +93,64 @@ func RegisterSubscriber(scheme string, factory SubscriberFactory) {
 	}
 	subReg[scheme] = factory
 }
+
+// ----------------------------------------------------------------------------
+
+// InMemSubscriber is a subscriber, that consumes messages from seeded data.
+// It is useful mainly for testing.
+type InMemSubscriber struct {
+	mu   sync.Mutex
+	msgs map[string][]SubMessage
+}
+
+// NewInMemSubscriber returns new subscriber, that consumes messages from seeded data.
+func NewInMemSubscriber(messagesByTopic map[string][]SubMessage) *InMemSubscriber {
+	byTopic := make(map[string][]SubMessage, len(messagesByTopic))
+	for topic, msgs := range messagesByTopic {
+		byTopic[topic] = append(make([]SubMessage, 0, len(msgs)), msgs...)
```

Comment:
nit: `byTopic[topic] = msgs` is fine, you are not mutating the original `msgs` slice throughout operation

Reply:
Addressed in https://github.com/bsm/bps/pull/6